src/memray/_memray/inject.cpp

```diff
@@ -0,0 +1,245 @@
+#define Py_LIMITED_API 0x03070000
+#include "Python.h"
+
+#include <netdb.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include <iostream>
+
+namespace memray {
+namespace {  // unnamed
+
+int
+connect_client(int port)
+{
+    struct addrinfo hints = {};
+    struct addrinfo* all_addresses = nullptr;
+    struct addrinfo* curr_address = nullptr;
+    int rv;
+    int sockfd;
+
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+
+    std::string port_str = std::to_string(port);
+    if ((rv = ::getaddrinfo(nullptr, port_str.c_str(), &hints, &all_addresses)) != 0) {
```

Comment:
This is fine, but I would suggest moving this out of the if because at first glance looks like a typo (= instead of ==). It may be easier to debug in gdb as well. Your call 👍 

If you do that you can move the declaration of `rv` here instead of at the top and make it `const`.